components/server/ee/src/prebuilds/bitbucket-server-service.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { RepositoryService } from "../../../src/repohost/repo-service";
+import { ProviderRepository, User } from "@gitpod/gitpod-protocol";
+import { inject, injectable } from "inversify";
+import { BitbucketServerApi } from "../../../src/bitbucket-server/bitbucket-server-api";
+import { AuthProviderParams } from "../../../src/auth/auth-provider";
+import { BitbucketServerContextParser } from "../../../src/bitbucket-server/bitbucket-server-context-parser";
+import { Config } from "../../../src/config";
+import { TokenService } from "../../../src/user/token-service";
+import { BitbucketServerApp } from "./bitbucket-server-app";
+
+@injectable()
+export class BitbucketServerService extends RepositoryService {
+    static PREBUILD_TOKEN_SCOPE = "prebuilds";
+
+    @inject(BitbucketServerApi) protected api: BitbucketServerApi;
+    @inject(Config) protected readonly config: Config;
+    @inject(AuthProviderParams) protected authProviderConfig: AuthProviderParams;
+    @inject(TokenService) protected tokenService: TokenService;
+    @inject(BitbucketServerContextParser) protected contextParser: BitbucketServerContextParser;
+
+    async getRepositoriesForAutomatedPrebuilds(user: User): Promise<ProviderRepository[]> {
+        const repos = await this.api.getRepos(user, { limit: 100, permission: "REPO_ADMIN" });
```

Comment:
[nit] Would be good to comment that we're requesting REPO_ADMIN because that's the minimum role required to create webhooks.